test/Conversion/VectorToAIEVec/test_redundant_store.mlir

```diff
@@ -0,0 +1,59 @@
+// RUN: aie-opt %s -affine-super-vectorize="virtual-vector-size=16" --convert-vector-to-aievec="aie-target=aieml" -canonicalize | FileCheck %s
+module {
+  func.func @matmul(%arg0: memref<64x64xi32>, %arg1: memref<64x64xi32>, %arg2: memref<64x64xi32>) {
+    affine.for %arg3 = 0 to 64 {
+      affine.for %arg4 = 0 to 64 {
+        affine.for %arg5 = 0 to 64 step 16 {
+          %0 = affine.load %arg0[%arg3, %arg5] : memref<64x64xi32>
+          %1 = affine.load %arg1[%arg5, %arg4] : memref<64x64xi32>
+          %2 = arith.muli %0, %1 : i32
+          %3 = affine.load %arg2[%arg3, %arg4] : memref<64x64xi32>
+          %4 = arith.addi %3, %2 : i32
+          affine.store %4, %arg2[%arg3, %arg4] : memref<64x64xi32>
+          %5 = affine.load %arg0[%arg3, %arg5 + 1] : memref<64x64xi32>
+          %6 = affine.load %arg1[%arg5 + 1, %arg4] : memref<64x64xi32>
+          %7 = arith.muli %5, %6 : i32
+          %8 = affine.load %arg2[%arg3, %arg4] : memref<64x64xi32>
+          %9 = arith.addi %8, %7 : i32
+          affine.store %9, %arg2[%arg3, %arg4] : memref<64x64xi32>
+          %10 = affine.load %arg0[%arg3, %arg5 + 2] : memref<64x64xi32>
+          %11 = affine.load %arg1[%arg5 + 2, %arg4] : memref<64x64xi32>
+          %12 = arith.muli %10, %11 : i32
+          %13 = affine.load %arg2[%arg3, %arg4] : memref<64x64xi32>
+          %14 = arith.addi %13, %12 : i32
+          affine.store %14, %arg2[%arg3, %arg4] : memref<64x64xi32>
+        }
+      }
+    }
+    return
+  }
+}
+
```

Comment:
As a test for "redundant store elimination", this test is too complex and probably in the wrong place. This should be testable on its own, and since it's not really testing "vector to aievec", probably living in the Dialect test suite. Of course, we should also test this end-to-end, but those should live in the Integration test suite, and this test (gemm) is probably already there.